src/instana/instrumentation/aws/boto3.py

```diff
@@ -0,0 +1,136 @@
+
+    @wrapt.patch_function_wrapper("botocore.auth", "SigV4Auth.add_auth")
+    def emit_add_auth_with_instana(
+        wrapped: Callable[..., None],
+        instance: "SigV4Auth",
+        args: Tuple[object],
+        kwargs: Dict[str, Any],
+    ) -> Callable[..., None]:
+        current_span = get_current_span()
+        if not tracing_is_off() and current_span and current_span.is_recording():
+            extract_custom_headers(current_span, args[0].headers)
+        return wrapped(*args, **kwargs)
+
+    @wrapt.patch_function_wrapper("botocore.client", "BaseClient._make_api_call")
+    def make_api_call_with_instana(
+        wrapped: Callable[..., Dict[str, Any]],
+        instance: Type["BaseClient"],
+        args: Sequence[Dict[str, Any]],
+        kwargs: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        # If we're not tracing, just return
+        if tracing_is_off():
+            return wrapped(*args, **kwargs)
+
+        tracer, parent_span, _ = get_tracer_tuple()
+
+        parent_context = parent_span.get_span_context() if parent_span else None
+
+        try:
+            if instance.meta.service_model.service_name == "dynamodb":
+                create_dynamodb_span(wrapped, instance, args, kwargs, parent_context)
+            elif instance.meta.service_model.service_name == "s3":
+                create_s3_span(wrapped, instance, args, kwargs, parent_context)
+            else:
+                with tracer.start_as_current_span(
+                    "boto3", span_context=parent_context
+                ) as span:
+                    try:
+                        operation = args[0]
+                        payload = args[1]
+
+                        span.set_attribute("op", operation)
+                        span.set_attribute("ep", instance._endpoint.host)
+                        span.set_attribute("reg", instance._client_config.region_name)
+
+                        span.set_attribute(
+                            SpanAttributes.HTTP_URL,
+                            instance._endpoint.host + ":443/" + args[0],
+                        )
+                        span.set_attribute(SpanAttributes.HTTP_METHOD, "POST")
+
+                        # Don't collect payload for SecretsManager
+                        if not hasattr(instance, "get_secret_value"):
+                            span.set_attribute("payload", payload)
+
+                        # Inject context when invoking lambdas
+                        if (
+                            "lambda" in instance._endpoint.host
+                            and operation == "Invoke"
+                        ):
+                            lambda_inject_context(payload, span)
+
+                    except Exception:
+                        logger.debug(
+                            "make_api_call_with_instana: collect error",
+                            exc_info=True,
+                        )
```

Comment:
This `try-except` looks unnecessary since you only set span attributes in this try block.
I suggest you have only one try-except block, keeping only the last one that executes the `wrapped()` function.